addons/l10n_in/models/account_invoice.py

```diff
@@ -67,6 +68,26 @@ def _compute_l10n_in_state_id(self):
             else:
                 move.l10n_in_state_id = False
 
+    @api.depends('l10n_in_state_id')
+    def _compute_fiscal_position_id(self):
+        other_country = None
+        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_invoice(include_receipts=True))
```

Comment:
`is_invoice` is for both sales and purchase documents
```suggestion
        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_sale_document(include_receipts=True))
```